src/main/java/de/tum/in/www1/artemis/domain/Exercise.java

```diff
@@ -96,6 +94,10 @@
     @Column(name = "difficulty")
     private DifficultyLevel difficulty;
 
+    @Enumerated(EnumType.STRING)
+    @Column(name = "mode")
+    private ExerciseMode mode;
```

Comment:
I could imagine a couple of other exercises modes that might come up in the future, so I picked an enum to be on the safe side, also since the downside is pretty low in my opinion.

Possible (but probably far out) scenarios for other exercise modes:
- `supervised` (each student or team gets a tutor assigned as a mentor)
- `public` (students work individually but submissions can be seen by everyone)